libvast/src/system/source.cpp

```diff
@@ -166,18 +182,42 @@ source(caf::stateful_actor<source_state>* self, format::reader_ptr reader,
       self->send(self->state.accountant, "source.start", now);
     },
     // get next element
-    [self](caf::unit_t&, caf::downstream<detail::framed<table_slice>>& out,
+    [self](caf::unit_t&, caf::downstream<detail::framed<table_slice>>&,
            size_t num) {
-      VAST_DEBUG("{} tries to generate {} messages", self, num);
+      if (self->state.has_sink && self->state.mgr->out().num_paths() == 0) {
+        VAST_WARN("{} discards request for {} messages because all its "
+                  "outbound paths were removed",
+                  self, num);
+        return;
+      }
+      VAST_DEBUG("{} schedules generation of {} messages", self, num);
+      self
+        ->request(caf::actor_cast<source_actor>(self), caf::infinite,
+                  atom::internal_v, atom::run_v, static_cast<uint64_t>(num))
+        .then(
+          [=]() {
+            VAST_DEBUG("{} finished generation of {} messages", self, num);
+          },
+          [=](const caf::error& err) {
+            VAST_WARN("{} failed generation of {} messages: {}", self, num,
+                      err);
+          });
```

Comment:
Not sure if the response really adds anything here, wouldn't the run-handler itself already print a warning if it encounters errors? On the other hand, it probably doesn't hurt either, except for readability.

Reply:
I don't expect this to ever be triggered, but now that it's in a separate handler having this is good practice. Otherwise, errors will go unreported should we ever decide to add one.